packages/translations/src/index.ts

```diff
@@ -22,49 +22,48 @@ import type {
  * @returns boolean which indicates if the languages are loaded
  */
 export function useTranslations(initParams: TranslationInitParams): boolean {
-  const [initialized, setInitialized] = useState(i.isInitialized)
-
-  !initialized && initTranslations(initParams)
+  const [loaded, setLoaded] = useState(false)
 
   useEffect(() => {
-    const handleInitialized = () => setInitialized(true)
-
-    // Check if initialized between render and useEffect
-    if (i.isInitialized) return handleInitialized()
-
-    // Listen if initialization takes longer
-    i.on('initialized', handleInitialized)
-    return () => i.off('initialized', handleInitialized)
+    initTranslations({
+      onDone: () => {
+        setLoaded(true)
+      },
+      ...initParams,
+    })
   }, [])
-
-  return initialized
+  return loaded
 }
 
 export function initTranslations({
   expirationTime,
   fetchOptions,
+  onDone,
   ...options
-}: TranslationInitParams) {
-  i.use(ChainedBackend).init({
-    ns: 'app',
-    defaultNS: 'app',
-    fallbackLng: options.lng || 'en',
-    load: 'all',
-    keySeparator: false,
-    nsSeparator: false,
-    interpolation: {
-      prefix: '[',
-      suffix: ']',
-    },
-    backend: {
-      backends: [StorageBackend, MultiloadAdapter],
-      backendOptions: [
-        { prefix: 'i18n_', expirationTime },
-        { backend: Fetch, backendOption: fetchOptions },
-      ],
+}: TranslationInitParams & { onDone: () => void }) {
```

Comment:
Why did you go for callback over a promise?

Reply:
no particular reason, just a preference. Is this an issue?

Reply:
Callback vs Promise: just a preference to not invent more options if i18next covers this already. I think returning a promise will need less code and is more idiomatic.

On the other hand, I was thinking of [i18next docs](https://www.i18next.com/overview/api#init):
> Do not call init multiple times.
To change language use [changeLanguage](https://github.com/overview/api#changelanguage). If you need complete different configs use [createInstance](https://github.com/overview/api#createinstance) or [cloneInstance](https://github.com/overview/api#cloneinstance).

And also thinking of [strict mode](https://github.com/reactwg/react-18/discussions/19) which may run useEffect twice in development.

My suggestion with `useState(() => {})` may also run twice, so is also not a solution.

Reply:
i18next exports a property `isInitialized` i can use this to check if i should initialize i18next or not